### ISSUE TITLE
Using a Minimizer Class for Confidence Level Calculations

### ISSUE THREAD

**Issue by kgspears** [lmfit/lmfit-py#360]
Hello Matt (and other experts).
My v8.3 code needed changing to v9.4 under python 3.5.  The fitting problem is a variable projection algorithm with 7-20 non-linear parameters and thousands of linear parameters done with matrix math in this algorithm.

I do consecutive fitting from Nelder to leastsq as follows.  Under v8.3 I could fit very quickly and do confidence intervals.  I cannot figure out how to use the Minimizer class with my residual function to do any fitting (and therefore cannot use it in the confidence interval calc, which is now required) 

Working code:  The residual function is quite complex for my variable projection algorithm; with many arrays passed to other functions (t (linear), data(2D), modlist (linear), wtfit(2D).

par0 = lmfit.Parameters()
result1 = lmfit.minimize(res, par0, args=(t, data, modlist, wtfit), method='nelder')
par1=result1.params
print(lmfit.fit_report(result1))
lskws = dict(ftol=1.e-4, xtol=1.e-4)  #--okay for most cases
result2 = lmfit.minimize(res, par1, args=(t, data, modlist, wtfit), method='leastsq', **lskws)
par2=result2.params

(where I use the par2 and result2 in lots of plots, writing files and hopefully confidence intervals)

Following the advanced example in confidence intervals and the info under Minimizer class (which I have never tried for this problem) it appears that I cannot define a suitable Minimizer
I tried a few things, one gave no errors
mini=lmfit.Minimizer(res, par2, args=(t, data, modlist, wtfit))   which gave no errors.
however, I cannot run a fit with the mini defined as above, so it must be incompatible in some way.  
result3=mini.minimize(method='leastsq')     
result3=mini.minimize(method='leastsq', params=par2)
Neither of these two ways to get a result3 worked, and had a similar error.

Here is the run and traceback references
mini=lmfit.Minimizer(res, par2, args=(t, data, modlist, wtfit))
(no error)
result3=mini.minimize(method='leastsq', params = par2)
Traceback (most recent call last):

  File "<ipython-input-82-618b579cd5ac>", line 1, in <module>
    result3=mini.minimize(method='leastsq', params = par2)

  File "C:\Users\ken\Anaconda3\lib\site-packages\lmfit\minimizer.py", line 1168, in minimize
    return function(**kwargs)

  File "C:\Users\ken\Anaconda3\lib\site-packages\lmfit\minimizer.py", line 1008, in leastsq
    lsout = scipy_leastsq(self.__residual, vars, **lskws)

  File "C:\Users\ken\Anaconda3\lib\site-packages\scipy\optimize\minpack.py", line 377, in leastsq
    shape, dtype = _check_func('leastsq', 'func', func, x0, args, n)

  File "C:\Users\ken\Anaconda3\lib\site-packages\scipy\optimize\minpack.py", line 26, in _check_func
    res = atleast_1d(thefunc(*((x0[:numinputs],) + args)))

TypeError: __residual() takes from 2 to 3 positional arguments but 6 were given

This looks like some syntax error in arguments using the Minimizer class that I cannot understand with my level of programming skill.

Thanks for your help.
Ken Spears

**Comment by newville** [comment]
@kgspears   If you want help, you need to follow the instructions. 

Please explain how it is possible to raise this as a github Issue after having read both https://github.com/lmfit/lmfit-py/blob/master/.github/CONTRIBUTING.md and https://github.com/lmfit/lmfit-py/blob/master/.github/ISSUE_TEMPLATE.md  as is clearly instructed to do so when creating a new issue?  I believe you had to have deliberately removed the text of ISSUE_TEMPLATE.md in order to post this Issue.   Those instructions say pretty clearly to NOT DO THIS and give details about how to ask for help, including which forum to use.   Was there something in those instructions that was not clear?

I find it difficult to understand how one could possible read all that and stilo post a question as a Github Issue.  I'm sure you will understand that we do not feel obligated to spend our free time helping rude people.

**Comment by newville** [comment]
Not an issue.

**Comment by kgspears** [comment]
Matt,
  I apologize for the message.  I was not being deliberately rude, but 
succeeded nonetheless!

   You spend a lot of time helping people, which everyone appreciates.  
I got involved documenting the problem and forgot that the google groups 
email system is the new way to call initial attention to an issue for 
vetting and possible help.

    It would be nice to continue using the confidence level 
calculation.  Since my residual functions are too complex, I will test 
this idea by converting your simple working example (which I tested 
prior) into a slightly more complex case, where the residual also uses 
extra arguments.  I will communicate the result via google groups if the 
problem persists, as it will be simpler for duplication.  There might be 
a real issue rather than unknown user error?

Ken
